src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -202,6 +202,11 @@ private static DateTime GetRawDateTimeConstant(CustomAttributeData attr)
 
         private object? GetDefaultValue(bool raw)
         {
+            if (IsRetval)
+            {
+                return DBNull.Value;
```

Comment:
Could not think of any case where return parameter could have a default value, double checking with the area experts CC @AaronRobinsonMSFT @jkotas @steveharter

Reply:
Tagging mono folks @lambdageek @marek-safar @vargaz

Reply:
Why is the extra if needed?

Reply:
Because GetDefaultValue returns null instead of DBNull.Value/Type.Missing, which makes HasDefaultValue return true for return parameters. Do we consider return parameter in mono to have default values for a valuable reason or it is just an error ?

Reply:
I think better fix would be to update [add_parameter_object_to_array](https://github.com/dotnet/runtime/blob/2a75e65fe2439071a17ae012b8d3e2baf0e49f7a/src/mono/mono/metadata/reflection.c#L981)

Reply:
I don't think it would fix the probem. The [add_parameter_object_to_array](https://github.com/dotnet/runtime/blob/2a75e65fe2439071a17ae012b8d3e2baf0e49f7a/src/mono/mono/metadata/reflection.c#L981) processes only input parameters. In fact, The ReturnParameter is constructed here :

https://github.com/dotnet/runtime/blob/bc766529c0eae0d51d3740dc43a49caa0c213d31/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.Mono.cs#L135-L138

Mono does not provide DefaultValueImpl to RuntimeParameterInfo as you can see, we can add a new method in mono like get_retval_marshal that return a hardcoded DbNull.Value or do it directly in the dedicated constructor:

https://github.com/dotnet/runtime/blob/bc766529c0eae0d51d3740dc43a49caa0c213d31/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs#L113-L122

Let me know what do you prefer